src/main/java/io/r2dbc/h2/client/SessionClient.java

```diff
@@ -48,21 +48,29 @@
 
     private final SessionInterface session;
 
+    private final boolean shutdownDatabaseOnClose;
+
     /**
      * Creates a new instance.
      *
      * @param connectionInfo the connection info to use
      * @throws NullPointerException if {@code connectionInfo} is {@code null}
      */
-    public SessionClient(ConnectionInfo connectionInfo) {
+    public SessionClient(ConnectionInfo connectionInfo, boolean shutdownDatabaseOnClose) {
         Assert.requireNonNull(connectionInfo, "connectionInfo must not be null");
 
         this.session = new SessionRemote(connectionInfo).connectEmbeddedOrServer(false);
+        this.shutdownDatabaseOnClose = shutdownDatabaseOnClose;
     }
 
     @Override
     public Mono<Void> close() {
         return Mono.defer(() -> {
+
+            if (this.shutdownDatabaseOnClose) {
+                CommandInterface shutdown = this.session.prepareCommand("SHUTDOWN", 0);
+                shutdown.executeUpdate(null);
+            }
```

Comment:
Do you think we need to guard against H2 exceptions with this command, and convert to R2BDC exceptions?